src/api/gpu/ucx/nixl_device.cuh

```diff
@@ -63,7 +63,15 @@ struct nixlGpuXferReqParams {
  */
 __device__ inline nixl_status_t
 nixlGpuConvertUcsStatus(ucs_status_t status) {
```

Comment:
The goal of this function was to use minimal branches for translating from ucx status to NIXL in data path.
So IMO it should be:
`return status == UCS_INPROGRESS ? NIXL_IN_PROG: NIXL_ERR_BACKEND;`

And you should keep `nixlGpuGetXferStatus` as is.